pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2256,42 +2256,42 @@ public void checkGC() {
-                            .thenAccept((r) -> {
-                        deleteMetadataFuture.complete(null);
-                    }).exceptionally(ex -> {
-                        deleteMetadataFuture.completeExceptionally(ex.getCause());
-                        return null;
-                    }))
-                    .exceptionally((e) -> {
-                        if (!(e.getCause() instanceof UnsupportedOperationException)) {
-                            log.error("delete metadata fail", e);
-                        }
-                        deleteMetadataFuture.complete(null);
-                        return null;
-                    });
-            return deleteMetadataFuture;
-        } catch (Exception e) {
-            return FutureUtil.failedFuture(e);
+        PartitionedTopicResources partitionedTopicResources = getBrokerService().pulsar().getPulsarResources()
+                .getNamespaceResources()
+                .getPartitionedTopicResources();
+        if (!topicName.isPartitioned()) {
+            return CompletableFuture.completedFuture(null);
         }
+        return partitionedTopicResources.partitionedTopicExistsAsync(topicName)
+                .thenCompose(partitionedTopicExist -> {
+                    if (!partitionedTopicExist) {
+                        return CompletableFuture.completedFuture(null);
+                    }
+                    return getBrokerService()
+                            .fetchPartitionedTopicMetadataAsync(TopicName.get(topicName.getPartitionedTopicName()))
+                            .thenCompose((metadata -> {
+                                List<CompletableFuture<Boolean>> persistentTopicExists =
+                                        new ArrayList<>(metadata.partitions);
+                                for (int i = 0; i < metadata.partitions; i++) {
+                                    persistentTopicExists.add(brokerService.getPulsar()
+                                            .getPulsarResources().getTopicResources()
+                                            .persistentTopicExists(topicName.getPartition(i)));
+                                }
+                                return FutureUtil.waitForAll(persistentTopicExists)
+                                        .thenAccept(__ -> {
+                                            // make sure all sub partitions were deleted after all future complete
+                                            persistentTopicExists.forEach(future -> {
+                                                boolean topicExist = future.join();
+                                                if (topicExist) {
+                                                    UnsupportedOperationException ex =
+                                                            new UnsupportedOperationException();
+                                                    log.error("[{}] Delete metadata fail", topic, ex);
```

Comment:
Don't need to print the log multiple times,

Reply:
Fixed. Thanks for your review.